virtio-devices/src/block.rs

```diff
@@ -660,6 +661,7 @@ impl Block {
                     physical_block_exp,
                     min_io_size: (topology.minimum_io_size / logical_block_size) as u16,
                     opt_io_size: (topology.optimal_io_size / logical_block_size) as u32,
+                    seg_max: (queue_size - 2) as u32,
```

Comment:
Can you please explain why the `seg_max` is set to this value?

Reply:
A request is consist of at least one our header and one in header, IIRC. What's left in the queue can be used for data segments.

Reply:
Thank you for the explanation. Can you please share some pointers for a more detailed context? I always find virtio spec way too concise to understand by itself.

Reply:
I looked at QEMU code. It was always like that since the beginning with no explanation.

The closest I can find is `virtblk_add_req` in Linux.